gcovr/__main__.py

```diff
@@ -188,206 +190,211 @@ def main(args=None):
 
     # load the config
     cfg_name = find_config_name(cli_options)
-    cfg_options = {}
-    if cfg_name is not None:
-        with io.open(cfg_name, encoding="UTF-8") as cfg_file:
-            cfg_options = parse_config_into_dict(
-                parse_config_file(cfg_file, filename=cfg_name)
-            )
-    options = merge_options_and_set_defaults([cfg_options, cli_options.__dict__])
-    # Reconfigure the logging.
-    if options.gcov_parallel > 1:
-        switch_to_logging_format_with_threads()
-
-    if options.verbose:
-        LOGGER.setLevel(logging.DEBUG)
-
-    if options.sort_branches and options.sort_key not in [
-        "uncovered-number",
-        "uncovered-percent",
-    ]:
-        LOGGER.error(
-            "the options --sort-branches without '--sort uncovered-number' or '--sort uncovered-percent' doesn't make sense."
-        )
-        sys.exit(EXIT_CMDLINE_ERROR)
-
-    if options.html_title == "":
-        LOGGER.error("an empty --html-title= is not allowed.")
-        sys.exit(EXIT_CMDLINE_ERROR)
-
-    for postfix in ["", "line", "branch"]:
-        key_medium = "html_medium_threshold"
-        key_high = "html_high_threshold"
-        if postfix:
-            key_medium += f"_{postfix}"
-            key_high += f"_{postfix}"
-        option_medium = f"--{key_medium.replace('_', '-')}"
-        option_high = f"--{key_high.replace('_', '-')}"
-
-        if getattr(options, key_medium) == 0:
-            LOGGER.error(f"value of {option_medium}= should not be zero.")
-            sys.exit(EXIT_CMDLINE_ERROR)
-
-        # Inherit the defaults from the global covarage values if not set
-        if postfix:
-            if getattr(options, key_medium) is None:
-                setattr(
-                    options,
-                    key_medium,
-                    options.html_medium_threshold,
-                )
-                # To get the correct option in the error message below.
-                option_medium = "--html-medium-threshold"
-            if getattr(options, key_high) is None:
-                setattr(
-                    options,
-                    key_high,
-                    options.html_high_threshold,
-                )
-                # To get the correct option in the error message below.
-                option_medium = "--html-high-threshold"
-
-        if getattr(options, key_medium) > getattr(options, key_high):
+    if cfg_name is None:
+        cfg_options_list = [{}]
+    else:
+        if os.path.splitext(cfg_name)[1].lower() == ".toml":
+            with io.open(cfg_name, "rb") as cfg_file:
+                cfg_options_list = parse_toml_config_file(cfg_file, filename=cfg_name)
+        else:
+            with io.open(cfg_name, encoding="UTF-8") as cfg_file:
+                cfg_options_list = parse_config_file(cfg_file, filename=cfg_name)
+
+    for cfg_options in cfg_options_list:
+        options = merge_options_and_set_defaults([cfg_options, cli_options.__dict__])
```

Comment:
This was because of the suggestion in https://github.com/gcovr/gcovr/issues/875#issuecomment-1912256805 but I'm happy to revert the change for support of multiple section because it's wired and changes the architecture.